### PR TITLE
integration: refactor regression test framework

### PR DESCRIPTION
**Pull request by JFixby**
The purpose of the PR is to implement regression testing framework for `btcd`, `dcrd` and `dcrwallet`.

Proposed changes:
 - Refactored package: `integration/`
Provides registry to ensure proper tracking and disposal of leaky assets.
Provides helpers for the test setup to detect its own defects.
Offers temporary directories management.
Offers management for reusable resources (e.g harnesses).
Test files were moved to `integration/harness/simpleregtest/`

 - Added package: `integration/commandline/`
Provides helpers to run external command-line tools, e.g `go` code builder, `btcd` and `btcwallet` executables.
Ensures proper disposal of the external processes to avoid CPU leaks.
See `example_test.go` for the usage.

 - Added package: `integration/gobuilder/`
Helps to build a target Go project upon request.

- `integration/rpctest/`-package was completely demolished.
Tests were split into separated files and moved to `integration/harness/simpleregtest/`
Harnesses and test setup code was refactored and moved to `integration/harness`

- Added package: `integration/harness/`
Provides reusable modules allowing to assemble and to configure any desired test environment.
Defines test harness and it's sub-components.

- Added package: `integration/harness/memwallet/`
Offers a simple in-memory wallet whose purpose is to provide basic test wallet functionality to the harness.

- Added package: `integration/harness/testnode/`
Offers a default test node that launches a new `btcd`-instance using command-line call.

- Added package: `integration/harness/simpleregtest/`
Harbours original tests from the `rpctest`-package and offers test setup to run these tests.

Tests:
image

### CODE REVIEW

**Comment by Roasbeef** [comment]
Are you aware of the `rpctest` pacakge?

**Comment by JFixby** [comment]
Travis successfully built and passed integration tests.

**Comment by JFixby** [comment]
I'll check it again and submit for review around next week.

**Comment by JFixby** [comment]
@Roasbeef this should be good to start the review process.

File: integration/gobuilder/util.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	"strings"
+
+	"github.com/btcsuite/btcd/integration"
+)
+
+// DetermineProjectPackagePath is used to determine target project path
+// for the following execution of a Go builder.
+// Starts from a current working directory, climbs up to a parent folder
+// with the target name and returns its path as a result.
+func DetermineProjectPackagePath(projectName string) string {
```
Comment (JFixby) [inline integration/gobuilder/util.go]: Addresses the https://github.com/btcsuite/btcd/commit/f5b60801a5a706984571bdc25cc228be793f61e0

**Comment by JFixby** [comment]
FYI: The next version of this PR is released here (as a separated repo): https://github.com/JFixby/btcregtest

**Comment by jakesyl** [comment]
@jcvernaleo (as per #1530)
- Low priority
- Testing
- Outdated

**Comment by guggero** [comment]
Not sure why GitHub closed this upon merge of another PR. Will try re-opening later, GitHub seems to have an issue currently.

**Comment by guggero** [comment]
Okay, I'm giving up. No idea why GitHub doesn't let me re-open the PR, sorry.

**Pull request closed without merging**